x-pack/plugins/apm/public/components/shared/Links/DiscoverLinks/DiscoverLink.tsx

```diff
@@ -5,20 +5,20 @@
  */
 
 import React from 'react';
-import { KibanaLink } from '../KibanaLink';
-import { QueryParamsDecoded } from '../url_helpers';
+import { KibanaRisonLink } from '../KibanaRisonLink';
+import { RisonAPMQueryParams } from '../rison_helpers';
 import { QueryWithIndexPattern } from './QueryWithIndexPattern';
 
 interface Props {
-  query: QueryParamsDecoded;
+  query: RisonAPMQueryParams;
   children: React.ReactNode;
 }
 
 export function DiscoverLink({ query, ...rest }: Props) {
   return (
     <QueryWithIndexPattern query={query}>
       {queryWithIndexPattern => (
-        <KibanaLink
+        <KibanaRisonLink
           pathname={'/app/kibana'}
           hash={'/discover'}
           query={queryWithIndexPattern}
```

Comment:
I think the separation is really good 👍 
This might be a bit too much for now, but I'm thinking we might be able to strip all the Rison logic even more down, so we don't need the rison_helper stuff. Pseudocode of the `DiscoverLink` component:

```js
export function DiscoverLink({
  query,
  children,
  urlParams,
  indexPattern
}: Props) {
  const risonEncoded = rison.encode({
    ...query,
    _g: {
      time: {
        from: urlParams.fromRange,
        to: urlParams.toRange
      }
    }
  });
  
  const search = risonEncoded // or something like it

  const href = url.format({
    pathname: chrome.addBasePath(pathname),
    hash: `${hash}?${search}`
  });

  return <EuiLink href={href} children={children} />;
}
```

Reply:
We can provide urlParams to `Discoverlink` by `connect`'ing it to redux.

Reply:
This actually works:
```js
export function DiscoverLink({ query, children }: Props) {
  // TODO: Remove hardcoded values for urlParams
  const rangeFrom = 'now-24h';
  const rangeTo = 'now';

  const queryEncoded = {
    _a: rison.encode(query._a),
    _g: rison.encode({
      ...query._g,
      time: {
        from: rangeFrom,
        to: rangeTo
      }
    })
  };

  const pathname = '/app/kibana';
  const search = `_a=${queryEncoded._a}&_g=${queryEncoded._g}`;
  const hash = '/discover';
  const href = getHref(pathname, hash, search);

  return <EuiLink href={href} children={children} />;
}

function getHref(pathname, hash, search) {
  return url.format({
    pathname: chrome.addBasePath(pathname),
    hash: `${hash}?${search}`
  });
}
```